flagengine/engine.go

```diff
@@ -1,115 +1,139 @@
 package flagengine
 
 import (
-	"github.com/Flagsmith/flagsmith-go-client/v4/flagengine/environments"
-	"github.com/Flagsmith/flagsmith-go-client/v4/flagengine/features"
-	"github.com/Flagsmith/flagsmith-go-client/v4/flagengine/identities"
-	"github.com/Flagsmith/flagsmith-go-client/v4/flagengine/identities/traits"
-	"github.com/Flagsmith/flagsmith-go-client/v4/flagengine/segments"
+	"fmt"
+
+	"github.com/Flagsmith/flagsmith-go-client/v5/flagengine/engine_eval"
+	"github.com/Flagsmith/flagsmith-go-client/v5/flagengine/utils"
 )
 
-// GetEnvironmentFeatureStates returns a list of feature states for a given environment.
-func GetEnvironmentFeatureStates(environment *environments.EnvironmentModel) []*features.FeatureStateModel {
-	if environment.Project.HideDisabledFlags {
-		var featureStates []*features.FeatureStateModel
-		for _, fs := range environment.FeatureStates {
-			if fs.Enabled {
-				featureStates = append(featureStates, fs)
-			}
-		}
-		return featureStates
-	}
-	return environment.FeatureStates
+// featureContextWithSegmentName holds a feature context along with the segment name it came from.
+type featureContextWithSegmentName struct {
+	featureContext *engine_eval.FeatureContext
+	segmentName    string
 }
 
-// GetEnvironmentFeatureState returns a specific feature state for a given featureName in a given environment, or nil feature state is not found.
-func GetEnvironmentFeatureState(environment *environments.EnvironmentModel, featureName string) *features.FeatureStateModel {
-	for _, fs := range environment.FeatureStates {
-		if fs.Feature.Name == featureName {
-			return fs
-		}
-	}
-	return nil
-}
+// GetEvaluationResult computes flags and matched segments given a context and a segment matcher.
+// The matcher should return true when the provided segment applies to the provided context.
+func GetEvaluationResult(ec *engine_eval.EngineEvaluationContext) engine_eval.EvaluationResult {
```

Comment:
Semi NIT: 
Overall I think we could abstract this one in all the SDKs, nothing changing functionally but it's basically doing 2 things:
- processSegments
- evaluate/process features.

Basically following your comments, imo it could be worth extracting them for better maintainability and unit tests. I know it's not a big impact either so your call

Reply:
Yeah, I was going to do that